pkgs/build-support/node/fetch-yarn-deps/index.js

```diff
@@ -40,12 +92,86 @@ const downloadFileHttps = (fileName, url, expectedHash, hashType = 'sha1') => {
 				if (expectedHash === undefined){
 					console.log(`Warning: lockfile url ${url} doesn't end in "#<hash>" to validate against. Downloaded file had hash ${h}.`);
 				} else if (h != expectedHash) return reject(new Error(`hash mismatch, expected ${expectedHash}, got ${h} for ${url}`))
-				resolve()
+				return resolve("");
```

Comment:
Could you explain what is the purpose of this change? Seems like it can be decoupled too.